openai/api_requestor.py

```diff
@@ -149,6 +151,63 @@ def format_app_info(cls, info):
             str += " (%s)" % (info["url"],)
         return str
 
+    def __check_polling_response(self, response: OpenAIResponse, predicate: Callable[[OpenAIResponse], bool]):
+        if predicate(response):
+            message, code = (None, None) if 'error' not in response.data else (
+                response.data['error']['message'] if 'message' in response.data['error'] else None,
+                response.data['error']['code'] if 'code' in response.data['error'] else None
+            )
+            message = message if message is not None else 'Operation failed'
+            raise error.OpenAIError(message=message, code=code)
+
+    def poll(
+        self,
+        method,
+        url,
+        until,
+        failed,
+        params = None,
+        headers = None,
+        interval = None,
+        delay = None
+    ) -> Tuple[Iterator[OpenAIResponse], bool, str]:
+        if delay:
+            time.sleep(delay)
+
+        response, b, api_key = self.request(method, url, params, headers)
+        self.__check_polling_response(response, failed)
+        while not until(response):
+            time.sleep(interval or response.retry_after or 1)
+            response, b, api_key = self.request(method, url)
+            self.__check_polling_response(response, failed)
+
+        response.data = response.data['result']
```

Comment:
Should we be capturing `created` from the response, too? https://platform.openai.com/docs/api-reference/images/create

Reply:
Talked with the service team and they will add created to result (in 2023-06-01-preview)